app/configurator/configurator-state.tsx

```diff
@@ -573,29 +594,27 @@ const transitionStepNext = (
           measures: dataSetMetadata.measures,
         });
 
-        const chartConfig = deriveFiltersFromFields(
-          getInitialConfig({
-            chartType: possibleChartTypes[0],
-            dimensions: dataSetMetadata.dimensions,
-            measures: dataSetMetadata.measures,
-          }),
-          dataSetMetadata.dimensions
-        );
+        const chartConfig = getInitialConfig({
+          chartType: possibleChartTypes[0],
+          dimensions: dataSetMetadata.dimensions,
+          measures: dataSetMetadata.measures,
+        });
+        deriveFiltersFromFields(chartConfig, dataSetMetadata.dimensions);
 
         return {
+          version: CONFIGURATOR_STATE_VERSION,
           state: "CONFIGURING_CHART",
           dataSet: draft.dataSet,
           dataSource: draft.dataSource,
           meta: draft.meta,
-          activeField: undefined,
-          chartConfig,
+          chartConfigs: [chartConfig],
```

Comment:
Can you try without replacing the array here ? 

chartConfigs.splice(0, chartConfig)

chartConfigs: chartConfigs

Reply:
Maybe

```
chartConfigs: chartConfigs.map(config => {
  if(config.key === chartConfig.key) { return chartConfig }
  return config
})
```

Reply:
Here the problem was that getter and setter were lost quite quickly, I think because of some spread pattern down the line, so this approach didn't work very well :( I think it's maybe even because of Immer, when it creates a copy of the object, we lose getters and setters

Reply:
Ah I see, and did you try without the getter / setter, but still with immer ?

Reply:
It turned out to work ok with replacing the config at given index. I guess the problem I had previously was related to something else :D